pkg/netpol/eval/internal/k8s/service.go

```diff
@@ -14,27 +14,48 @@
 package k8s
 
 import (
+	"errors"
+	"fmt"
+
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 // Service encapsulates k8s service fields that are relevant for connectivity analysis
 type Service struct {
 	Name      string
 	Namespace string
-	selectors map[string]string
+	Selectors map[string]string
 	// todo : support ports field?!
 }
 
 func ServiceFromCoreObject(svcObj *corev1.Service) (*Service, error) {
 	svc := &Service{
 		Name:      svcObj.Name,
 		Namespace: svcObj.Namespace,
-		selectors: make(map[string]string, len(svcObj.Spec.Selector)),
+		Selectors: make(map[string]string, len(svcObj.Spec.Selector)),
+	}
+
+	if svcObj.Spec.Selector == nil {
+		return nil, errors.New("K8s Service without selectors is not supported")
 	}
 
 	for k, v := range svcObj.Spec.Selector {
-		svc.selectors[k] = v
+		svc.Selectors[k] = v
 	}
 
 	return svc, nil
 }
+
+func (svc *Service) ServicSelectorsAsLabelSelector() (labels.Selector, error) {
+	labelsSelector := metav1.LabelSelector{MatchLabels: svc.Selectors}
+	selectorRes, err := metav1.LabelSelectorAsSelector(&labelsSelector)
+	if err != nil {
+		svcStr := types.NamespacedName{Namespace: svc.Namespace, Name: svc.Name}
+		return nil, fmt.Errorf(" service %s : %s", svcStr, selectorErrTitle)
```

Comment:
what is `selectorErrTitle`?

Reply:
a `const` declared in the `netpol.go `(same pkg) 

`selectorErrTitle  = "selector error"`